### PR TITLE
Add Login and MyTrails views

### PR DESCRIPTION
**Pull request by kotlarmilos**
## Description

This PR introduces Login and MyTrails views to feature AppleID auth and in-app purchase.

Screenshot 2024-10-18 at 18 55 00
Screenshot 2024-10-18 at 18 55 08
Screenshot 2024-10-18 at 18 56 47

### CODE REVIEW

File: HikingApp.MacCatalyst/ViewControllers/MainViewController.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        public override void ViewDidLoad()
+        {
+            base.ViewDidLoad();
+
+            var exploreViewController = new ExploreViewController();
+            var myTrailsViewController = new MyTrailsViewController();
+
+            var exploreNavigationController = new UINavigationController(exploreViewController)
+            {
+                TabBarItem = new UITabBarItem("Explore", UIImage.FromBundle("explore_icon.png"), 0)
```
Comment (matouskozak) [inline HikingApp.MacCatalyst/ViewControllers/MainViewController.cs]: Do we have the `explore_icon.png` and `mytrails_icon.png` somewhere?

Comment (kotlarmilos) [reply HikingApp.MacCatalyst/ViewControllers/MainViewController.cs]: No, let's remove them.

Comment (matouskozak) [inline HikingApp.MacCatalyst/ViewControllers/MainViewController.cs]:
```suggestion
                TabBarItem = new UITabBarItem("Explore", null, 0)
```

File: HikingApp.MacCatalyst/ViewControllers/MainViewController.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            };
+
+            var myTrailsNavigationController = new UINavigationController(myTrailsViewController)
+            {
+                TabBarItem = new UITabBarItem("My Trails", UIImage.FromBundle("mytrails_icon.png"), 1)
```
Comment (matouskozak) [inline HikingApp.MacCatalyst/ViewControllers/MainViewController.cs]:
```suggestion
                TabBarItem = new UITabBarItem("My Trails", null, 1)
```

**Review by matouskozak — APPROVED** [review]
LGTM! thanks for the upgrade.

**Pull request merged**